gtk4/src/subclass/widget.rs

```diff
@@ -181,18 +181,25 @@ impl<T: WidgetImpl> WidgetImplExt for T {
         unsafe {
             let data = T::type_data();
             let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkWidgetClass;
-            let f = (*parent_class)
-                .compute_expand
-                .expect("No parent class impl for \"compute_expand\"");
-            let mut h: i32 = hexpand_p.to_glib();
-            let mut v: i32 = vexpand_p.to_glib();
-            f(
-                widget.unsafe_cast_ref::<Widget>().to_glib_none().0,
-                &mut h,
-                &mut v,
-            );
-            *hexpand_p = from_glib(h);
-            *vexpand_p = from_glib(v);
+            let widget = widget.unsafe_cast_ref::<Widget>();
+            if let Some(f) = (*parent_class).compute_expand {
+                let mut h: i32 = hexpand_p.to_glib();
+                let mut v: i32 = vexpand_p.to_glib();
+                f(widget.to_glib_none().0, &mut h, &mut v);
+                if !widget.get_hexpand_set() {
+                    *hexpand_p = from_glib(h);
```

Comment:
You need to do something similar also in the function called from C, otherwise you'll pass around mutable references to uninitialized data if `Xexpand_set == false`.